modin/core/io/text/csv_glob_dispatcher.py

```diff
@@ -373,28 +386,29 @@ def get_path(cls, file_path: str) -> list:
         list
             List of strings of absolute file paths.
         """
-        if S3_ADDRESS_REGEX.search(file_path):
-            # S3FS does not allow captial S in s3 addresses.
-            if file_path[0] == "S":
-                file_path = "{}{}".format("s", file_path[1:])
-
-            S3FS = import_optional_dependency(
-                "s3fs", "Module s3fs is required to read S3FS files."
+        if fsspec.core.split_protocol(file_path)[0] in _SUPPORTED_PROTOCOLS:
+            # `file_path` may start with a capital letter, which isn't supported by `fsspec.core.url_to_fs` used below.
+            file_path = file_path[0].lower() + file_path[1:]
+
+            from botocore.exceptions import (
+                NoCredentialsError,
+                EndpointConnectionError,
+                ConnectTimeoutError,
             )
-            from botocore.exceptions import NoCredentialsError, EndpointConnectionError
 
             def get_file_path(fs_handle) -> List[str]:
                 file_paths = fs_handle.glob(file_path)
-                s3_addresses = ["{}{}".format("s3://", path) for path in file_paths]
+                if len(file_paths) == 0 and not fs_handle.exists(file_path):
+                    raise FileNotFoundError(f"Path <{file_path}> isn't available.")
+                s3_addresses = [f"s3://{path}" for path in file_paths]
```

Comment:
Should we change this to not assume the addresses are all for s3?

Reply:
For now we can leave as it is, because this block of code is used only in case s3 storage (`_SUPPORTED_PROTOCOLS` for read_csv_glob is {s3, S3} only).